PyInstaller/utils/win32/versioninfo.py

```diff
@@ -210,6 +210,13 @@ def toRaw(self):
         return (struct.pack('hhh', sublen, vallen, typ)
                 + raw_name + b'\000\000' + pad + rawffi + pad2 + tmp)
 
+    def __eq__(self, other):
+        if isinstance(other, VSVersionInfo):
+            return self.toRaw() == other.toRaw()
+        if isinstance(other, bytes):
+            return self.toRaw() == other
```

Comment:
You don't actually need to write all the cases:

```suggestion
        return self.toRaw() == other
```
If `self` and `other` are both `VSVersionInfo`s then `self == other` reduces to:
```python
self.toRaw() == other
```
To evaluate that, Python first tries:
```python
bytes.__eq__(self.toRaw(), other)
```
which fails because `bytes` doesn't know how to handle a `VSVersionInfo` but Python knows that `a == b` is equivalent to `b == a` so it tries the opposite way which does work.
```python
other == self.toRaw()
```
This calls `VSVersionInfo.__eq__(other, self.toRaw())`, eventually giving:
```python
other.toRaw() == self.toRaw()
```

Reply:
Clever! (I copied that method from `winmanifest`, which probably was written when Python 2 was still supported, and that trick wouldn't have worked then.)